### PR TITLE
K8SPG-437: merge upstream 5.4.2 changes

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPG-437](https://jira.percona.com/browse/K8SPG-437) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPG-437

**DESCRIPTION**
---
**Problem:**
*We need to merge upstream changes.*

**Solution:**
*Merge the changes.*

*`InstanceSidecars` feature gate is disabled by default after the changes. This feature gate is necessary for PMM to work. This is why we should force it to be enabled, even if it is disabled in `PGO_FEATURE_GATES` env var.*

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Are the manifests (crd/bundle) regenerated if needed?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported PG version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/256237ef6/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/256237ef6/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/256237ef6/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/256237ef6/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/256237ef6/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/256237ef6/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/256237ef6/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/256237ef6/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/256237ef6/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/256237ef6/users.log) |
| We run 10 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/518/commits/256237ef654a056dcee6c2f4f248f5809e042397
image: `perconalab/percona-postgresql-operator:PR-518-256237ef6`

File: installers/olm/bundle.annotations.yaml [hunk 1]
```diff
@@ -26,13 +26,13 @@ annotations:
   # the value from the bundle with the highest semantic version.
   #
   # https://olm.operatorframework.io/docs/best-practices/channel-naming/
-  operators.operatorframework.io.bundle.channels.v1: v2
-  operators.operatorframework.io.bundle.channel.default.v1: v2
+  operators.operatorframework.io.bundle.channels.v1: v5
+  operators.operatorframework.io.bundle.channel.default.v1: v5
```
Comment (egegunes) [inline installers/olm/bundle.annotations.yaml:30]: these should be v2

File: installers/olm/config/operator/kustomization.yaml [hunk 2]
```diff
@@ -3,3 +3,7 @@ kind: Kustomization
 
 resources:
 - ../../../../config/bundle
+#- ../../../../config/default
+#
+#patches:
+#- path: target-namespace.yaml
```
Comment (egegunes) [inline installers/olm/config/operator/kustomization.yaml:9]: please don't commit commented code

File: installers/olm/config/redhat/kustomization.yaml [hunk 3]
```diff
@@ -3,4 +3,7 @@ kind: Kustomization
 
 resources:
 - ../operator
-- ../examples
\ No newline at end of file
+- ../examples
+#
+#patches:
+#- path: related-images.yaml
```
Comment (egegunes) [inline installers/olm/config/redhat/kustomization.yaml:9]: no commented code

File: config/default/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+  postgres-operator.crunchydata.com/control-plane: postgres-operator
+
+bases:
+- ../crd
+- ../rbac/cluster
+- ../manager
+
+#patchesJson6902:
+#- target: { group: apps, version: v1, kind: Deployment, name: pgo }
+#  path: selectors.yaml
```
Comment (egegunes) [inline config/default/kustomization.yaml:16]: please don't commit commented code

**Review by egegunes — CHANGES_REQUESTED** [review]

**Review by inelpandzic — COMMENTED** [review]
@pooknull this looks good, just address Ege's comments and we can merge it.

File: installers/olm/config/redhat/kustomization.yaml (hunk 3 above)
Comment (pooknull) [reply installers/olm/config/redhat/kustomization.yaml]: https://github.com/percona/percona-postgresql-operator/pull/518/commits/a5fc9520a6dd3b0023091938b40ba45684f77bb6

File: installers/olm/config/operator/kustomization.yaml (hunk 2 above)
Comment (pooknull) [reply installers/olm/config/operator/kustomization.yaml]: https://github.com/percona/percona-postgresql-operator/pull/518/commits/a5fc9520a6dd3b0023091938b40ba45684f77bb6

File: installers/olm/bundle.annotations.yaml (hunk 1 above)
Comment (pooknull) [reply installers/olm/bundle.annotations.yaml]: https://github.com/percona/percona-postgresql-operator/pull/518/commits/a5fc9520a6dd3b0023091938b40ba45684f77bb6

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/a5fc9520a/demand-backup.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/a5fc9520a/init-deploy.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/a5fc9520a/monitoring.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/a5fc9520a/operator-self-healing.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/a5fc9520a/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/a5fc9520a/scheduled-backup.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/a5fc9520a/self-healing.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/a5fc9520a/start-from-backup.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/a5fc9520a/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-518/a5fc9520a/users.log) |
| We run 10 out of 10|

commit: https://github.com/percona/percona-postgresql-operator/pull/518/commits/a5fc9520a6dd3b0023091938b40ba45684f77bb6
image: `perconalab/percona-postgresql-operator:PR-518-a5fc9520a`

**Review by egegunes — APPROVED** [review]

**Review by inelpandzic — APPROVED** [review]

**Review by hors — APPROVED** [review]

**Pull request merged**